Re-syncing new commits into an existing repository result in inaccessible content

### Steps to reproduce the issue
Instantiate a new repository locally and publish it:
```
# on the host machine
ostree --repo=repo1 init --mode=archive
mkdir test
echo "test1" > test/test.txt
ostree --repo=repo1 commit --branch foo test
ostree summary -u --repo=repo1
```
```
# on the host machine
python3 -m http.server 8001
```
```
# on the pulp machine
pulp ostree repository create --name foo
pulp ostree remote create --name foo --url http://0.0.0.0:8001/repo1/
pulp ostree repository sync --name foo --remote foo
pulp ostree distribution create --name foo --base-path foo --repository foo
http http://localhost:5001/pulp/content/foo/summary
```

Add a new commit to the repository and re-sync it:
```
# on the host machine
echo "test2" > test/test2.txt
ostree --repo=repo1 commit --branch foo test
ostree summary -u --repo=repo1
```
```
# on the pulp machine
pulp ostree repository sync --name foo --remote foo
http http://localhost:5001/pulp/content/foo/summary
```

```
pulp [None]: pulpcore.content.handler:ERROR: Multiple (pass-through) matches for {b}/{p}
[2023-05-24 17:05:33 +0000] [1141] [ERROR] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/local/lib/python3.8/site-packages/pulpcore/content/authentication.py", line 48, in authenticate
    return await handler(request)
  File "/usr/local/lib/python3.8/site-packages/pulpcore/content/handler.py", line 245, in stream_content
    return await self._match_and_stream(path, request)
  File "/usr/local/lib/python3.8/site-packages/pulpcore/content/handler.py", line 673, in _match_and_stream
    ca = await sync_to_async(get_contentartifact_blocking)()
  File "/usr/local/lib/python3.8/site-packages/asgiref/sync.py", line 448, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "/usr/lib64/python3.8/asyncio/tasks.py", line 455, in wait_for
    return await fut
  File "/usr/lib64/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.8/site-packages/asgiref/sync.py", line 490, in thread_handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pulpcore/content/handler.py", line 668, in get_contentartifact_blocking
    ca = ContentArtifact.objects.select_related(
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 439, in get
    raise self.model.MultipleObjectsReturned(
pulpcore.app.models.content.ContentArtifact.MultipleObjectsReturned: get() returned more than one ContentArtifact -- it returned 2!
127.0.0.1 [24/May/2023:17:05:33 +0000] "GET /pulp/content/foo/summary HTTP/1.0" 500 225 "-" "HTTPie/3.2.2"
```

The errors are raised for config and summary files.

--- a/CHANGES/257.bugfix
+++ b/CHANGES/257.bugfix
@@ -1,0 +1,1 @@
+Fixed a bug that disallowed users to pull re-synced content locally.

--- a/pulp_ostree/app/models.py
+++ b/pulp_ostree/app/models.py
@@ -99,6 +99,7 @@ class OstreeConfig(Content):
     """A content model for an OSTree repository configuration file."""
 
     TYPE = "config"
+    repo_key_fields = ("relative_path",)
 
     sha256 = models.CharField(max_length=64, db_index=True)
     relative_path = models.TextField(null=False)
@@ -112,6 +113,7 @@ class OstreeSummary(Content):
     """A content model for an OSTree summary file."""
 
     TYPE = "summary"
+    repo_key_fields = ("relative_path",)
 
     sha256 = models.CharField(max_length=64, db_index=True)
     relative_path = models.TextField(null=False)

